### DISCUSSION TITLE
Sinilink PCIe Computer Remote (XY-WPCE)

### DISCUSSION THREAD

**Discussion by eoghan2t9** [arendst/Tasmota#14873] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [* ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [*] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [*] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [*] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [*] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [*] Device used (e.g., Sonoff Basic): Sinilink PCIe Computer Remote (XY-WPCE)
- [ ] Tasmota binary firmware version number used: 11.0.0
  - [*] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: Tazmotizer
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
09:47:01.197 CMD: Backlog Template; Module; GPIO 255;
09:47:01.271 MQT: stat/tasmota_FED32A/RESULT = {"NAME":"XY-WPCE","GPIO":[1,1,320,1,32,224,0,0,160,0,0,0,0,0],"FLAG":0,"BASE":18}
09:47:01.485 MQT: stat/tasmota_FED32A/RESULT = {"Module":{"0":"XY-WPCE"}}
09:47:01.736 MQT: stat/tasmota_FED32A/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"320":"Led_i1"},"GPIO3":{"0":"None"},"GPIO4":{"32":"Button1"},"GPIO5":{"224":"Relay1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"160":"Switch1"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
09:47:56.882 CMD: Status 0
09:47:56.890 MQT: stat/tasmota_FED32A/STATUS = {"Status":{"Module":0,"DeviceName":"PC Remote","FriendlyName":["PC Remote"],"Topic":"tasmota_FED32A","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[2,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
09:47:56.896 MQT: stat/tasmota_FED32A/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Power On","Uptime":"0T05:54:24","StartupUTC":"2022-02-16T02:53:32","Sleep":50,"CfgHolder":4617,"BootCount":17,"BCResetTime":"2022-02-15T23:04:21","SaveCount":88,"SaveAddress":"F4000"}}
09:47:56.904 MQT: stat/tasmota_FED32A/STATUS2 = {"StatusFWR":{"Version":"11.0.0(tasmota)","BuildDateTime":"2022-02-12T14:13:50","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"348/699"}}
09:47:56.912 MQT: stat/tasmota_FED32A/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":2,"LogHost":"","LogPort":514,"SSId":["VM3160719-IOT",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006000","00004001"]}}
09:47:56.927 MQT: stat/tasmota_FED32A/STATUS4 = {"StatusMEM":{"ProgramSize":620,"Free":380,"Heap":23,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
09:47:56.933 MQT: stat/tasmota_FED32A/STATUS5 = {"StatusNET":{"Hostname":"tasmota-FED32A-4906","IPAddress":"192.168.1.208","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.2","DNSServer2":"0.0.0.0","Mac":"D8:F1:5B:FE:D3:2A","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
09:47:56.939 MQT: stat/tasmota_FED32A/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.2","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_FED32A","MqttUser":"eoghan","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
09:47:56.948 MQT: stat/tasmota_FED32A/STATUS7 = {"StatusTIM":{"UTC":"2022-02-16T08:47:56","Local":"2022-02-16T09:47:56","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"07:56","Sunset":"18:11"}}
09:47:56.956 MQT: stat/tasmota_FED32A/STATUS10 = {"StatusSNS":{"Time":"2022-02-16T09:47:56","Switch1":"OFF"}}
09:47:56.965 MQT: stat/tasmota_FED32A/STATUS11 = {"StatusSTS":{"Time":"2022-02-16T09:47:56","Uptime":"0T05:54:24","UptimeSec":21264,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"VM3160719-IOT","BSSId":"60:38:E0:CE:3E:A9","Channel":11,"Mode":"11n","RSSI":100,"Signal":-38,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
09:49:33.278 CMD: Weblog 4
09:49:33.285 MQT: stat/tasmota_FED32A/RESULT = {"WebLog":4}
09:49:33.430 CFG: Saved to flash at FB, Count 89, Bytes 4096
09:49:35.393 WIF: Sending Gratuitous ARP
09:49:40.526 WIF: Checking connection...
09:49:41.331 HTP: Main Menu
09:49:43.566 SRC: WebGui from 192.168.1.101
09:49:43.572 MQT: stat/tasmota_FED32A/RESULT = {"POWER":"OFF"}
09:49:43.577 MQT: stat/tasmota_FED32A/POWER = OFF
09:49:44.359 CFG: Saved to flash at FA, Count 90, Bytes 4096
09:49:52.618 SRC: WebGui from 192.168.1.101
09:49:52.624 MQT: stat/tasmota_FED32A/RESULT = {"POWER":"ON"}
09:49:52.629 MQT: stat/tasmota_FED32A/POWER = ON
09:49:52.859 MQT: stat/tasmota_FED32A/RESULT = {"Switch1":{"Action":"ON"}}
09:49:53.349 CFG: Saved to flash at F9, Count 91, Bytes 4096
09:49:58.650 MQT: stat/tasmota_FED32A/RESULT = {"Switch1":{"Action":"OFF"}}
09:50:00.522 WIF: Checking connection...

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Install the PCIe card following the manufacture instruction's after flashing Tasmota, go to the Tasmota WebUI and toggle the power button. The PC will power on for a few seconds then power off again. Tasmota's WebUI doesn't reflect this and thinks the PC is still on. I have a feeling that the pins on the PCIe board that control the power sw header on the motherboard are being kept shorted for too long and the motherboard thinks I'm trying to force a shutdown. Also on the PCIe card there is a header for the case switch to be inserted so the user still has manual control however Tasmota doesn't seem to detect that this button is being used so it more or less may as well be disconnected completely and the Tasmota log doesn't even show this physical button being pressed.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
When the button is toggled within the Webui the PC should power on as it the physical power button was pressed. Case power button should also able to be used in the event of a PC lock up where a forced shutdown maybe needed.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

Sorry if I have filled this form incorrectly.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by eoghan2t9** [comment, marked as answer]
Just to add I just noticed that if I'm in the WebUI and toggle the button on and once the PC powers on I toggle the button off again the PC starts normally. However the case button still does nothing 😞

**Reply by sfromis** [reply]
https://discord.com/channels/479389167382691863/490244847568158751/943432004580560916

**Reply by BlueWings172** [reply]
@eoghan2t9 have you solved this ? I'm having the same issue:

1- Case power button doesn't do anything.
2- Tasmota relay status shows Off in all states.
3- I cannot shutdown my PC! when I click shutdown on Windows, the PC just restarts !

**Comment by danilogit** [comment]
Hi, did you find the solution for this issue? How do you put the XY-WPCE in flash mode to install Tasmota?

**Reply by BlueWings172** [reply]
@danilogit, I don't know if you solved this, but just in case you didn't, in order to put the device in flash mode you have to connect IO0 to Ground. So Ground from your flashing device should be connected to both XY-WPCE's Ground and IO0. Hope this helps.

**Reply by danilogit** [reply]
@BlueWings172 Hi! I did it. To enable power button you need to create a Tasmota Rule. This rule will be responsible for triggering the relay when you press a switch. I made all features as it is in Sinilink firmware using NodeJs, MQTT with [Mosquitto ](https://mosquitto.org/)and the template as shown in the image below. I wrote a Javascript backend app to deal with MQTT messages. Enter the rule below in your Tasmota console and power button should work correctly. 

`Rule1 ON switch2#state=1 DO Power1 1 ENDON ON switch2#state=0 DO Power1 0 ENDON`

My configuration template looks like this now:

![image](https://user-images.githubusercontent.com/5581367/179418584-e383d21e-d8bc-4957-9900-0d6b85cfd2d4.png)

GPIO10 should be set to receive the state of your PC, if it is ON or OFF.

Try it. I don`t know if I am forgetting something. If I remember what else I did  I update you.

**Reply by BlueWings172** [reply]
@danilogit 

Many thanks for sharing a solution for this issue! Will give a try and let you know.

**Comment by BlueWings172** [comment]
@sfromis 

The link you posted does not lead anywhere. Can you please repost it ?

**Reply by sfromis** [reply]
The link very much lead to somewhere, if you are on the Discord server for Tasmota. It was a reference to the issue being discussed there, and that it was effectively bad-form cross-posting to also open it here.

**Reply by BlueWings172** [reply]
@sfromis 

Many thanks. I was able to join the discord server and see the posts.

I'm a beginner and i'm not sure what needs to be done here. When I click the PC case button, nothing appears on the console, so I'm not sure how I can create a rule to toggle the relay if the button press does not register. Is the template missing something ?

Also what confuses me is that when I shutdown the PC in Windows, it just restarts now but when I toggle the relay from Tasmota, the PC turns off as it should.

I would appreciate any help or information you can provide. Thanks

**Reply by sfromis** [reply]
I have no experience with the device, I just supplemented with perspective from the Tasmota side. If you search for the user starting the discussion on Discord, you'll find later posts about the part with it restarting, but not exactly the same situation.

**Reply by BlueWings172** [reply]
This has been solved by @diegargon in [this post ](https://github.com/blakadder/templates/issues/1472#issuecomment-1186498092)

**Comment by rucanunes** [comment]
Hello,

Could you please tell me if you found how this works? I'm having the same problem.

Thank you!

**Reply by diegargon** [reply]
https://github.com/blakadder/templates/issues/1472#issuecomment-1186498092

**Reply by BlueWings172** [reply]
Have you tried this post https://github.com/blakadder/templates/issues/1472#issuecomment-1186498092

**Comment by speedycars** [comment]
For anyone ending their search here, I'm sharing my experience with XY-WPCE.
Don't know if my unit was defective, but it was never able to listen for the physical PC power events, otherwise the smart relay to power on/off the PC via Tasmota web interface works flawlessly.
Tried Tasmota with all templates I've found - none worked. Then tried it with ESPhome - exactly the same experience.
Turns out the GPIO10 on this module doesn't work properly, later on read that GPIO10 is a special pin and is expected to see such issues with it.
Fortunately, there is another GPIO available - GPIO4 which is used for the physical button mounted directly on circuit - originally used for network pairing.
What I did is to dismount this circuit button and solder 2 dupont cables on the ex-button circuit ends. Then move switch2 to GPIO4 instead of GPIO10 in Tasmota config.
This fixed it all and now this module works as expected.

Used configuration found on https://templates.blakadder.com/sinilink_XY-WPCE.html .
*Note the rule below to mimic the physical PC power button (now connected to the two newly soldered dupont cables) and that it is configured in two console command lines:
Rule1 ON switch2#state=1 DO Power1 1 ENDON ON switch2#state=0 DO Power1 0 ENDON
Rule1 1
Then just enable Switch2 on GPIO4 and disable (select "None") for GPIO10 in the configuration.
I hope it will be helpful to someone someday here.
